server/src/main/java/org/apache/iotdb/db/engine/compaction/task/CompactionRecoverHelper.java

```diff
@@ -0,0 +1,198 @@
+                  Long.parseLong(
+                      timePartitionDir
+                          .getPath()
+                          .substring(timePartitionDir.getPath().lastIndexOf(File.separator) + 1)),
+                  compactionLog,
+                  timePartitionDir.getPath(),
+                  isSequence,
+                  tsFileManager)
+              .call();
+        }
+      }
+    }
+
+    // search compaction log for old LevelCompaction
+    File logFile =
+        FSFactoryProducer.getFSFactory()
+            .getFile(
+                tsFileManager.getStorageGroupDir(),
+                logicalStorageGroupName + INNER_COMPACTION_LOG_NAME_SUFFIX_FROM_OLD);
+    if (logFile.exists()) {
+      IoTDBDescriptor.getInstance()
+          .getConfig()
+          .getInnerCompactionStrategy()
+          .getCompactionRecoverTask(
+              tsFileManager.getStorageGroupName(),
+              tsFileManager.getVirtualStorageGroup(),
+              -1,
+              logFile,
+              logFile.getParent(),
+              isSequence,
+              tsFileManager)
+          .call();
+    }
+  }
+
+  public void recoverCrossSpaceCompaction() throws Exception {
+    logger.info("recovering cross compaction");
+    recoverCrossCompactionFromOldVersion();
+    recoverCrossCompaction();
+    logger.info("try to synchronize CompactionScheduler");
+  }
+
+  private void recoverCrossCompaction() throws Exception {
+    List<String> sequenceDirs = DirectoryManager.getInstance().getAllSequenceFileFolders();
+    for (String dir : sequenceDirs) {
+      File storageGroupDir =
+          new File(
+              dir
+                  + File.separator
+                  + logicalStorageGroupName
+                  + File.separator
+                  + virtualStorageGroupId);
+      if (!storageGroupDir.exists()) {
+        return;
+      }
+      File[] timePartitionDirs = storageGroupDir.listFiles();
+      if (timePartitionDirs == null) {
+        return;
+      }
+      for (File timePartitionDir : timePartitionDirs) {
+        if (!timePartitionDir.isDirectory()
+            || !Pattern.compile("[0-9]*").matcher(timePartitionDir.getName()).matches()) {
+          continue;
+        }
+        File[] compactionLogs =
+            CompactionLogger.findCrossSpaceCompactionLogs(timePartitionDir.getPath());
+        for (File compactionLog : compactionLogs) {
+          logger.info("calling cross compaction task");
+          IoTDBDescriptor.getInstance()
+              .getConfig()
+              .getCrossCompactionStrategy()
+              .getCompactionRecoverTask(
+                  logicalStorageGroupName,
+                  virtualStorageGroupId,
+                  Long.parseLong(timePartitionDir.getName()),
+                  compactionLog,
+                  tsFileManager)
+              .call();
+        }
+      }
+    }
+  }
+
+  private void recoverCrossCompactionFromOldVersion() throws Exception {
```

Comment:
It's better to point out the specific version number in method name such `recoverCrossCompactionBelow013` and add comments for the method.

Reply:
Resolved